src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -846,5 +846,25 @@ internal PrimitiveColumnContainer<float> CloneAsFloatContainer()
             ret.NullCount = NullCount;
             return ret;
         }
+
+        internal PrimitiveColumnContainer<DateTime> CloneAsDateTimeContainer()
+        {
+            var ret = new PrimitiveColumnContainer<DateTime>();
+            foreach (ReadOnlyDataFrameBuffer<T> buffer in Buffers)
+            {
+                ret.Length += buffer.Length;
+                DataFrameBuffer<DateTime> newBuffer = new DataFrameBuffer<DateTime>();
+                ret.Buffers.Add(newBuffer);
+                newBuffer.EnsureCapacity(buffer.Length);
+                ReadOnlySpan<T> span = buffer.ReadOnlySpan;
```

Comment:
>T [](http://example.com/codeflow?start=29&length=1)

does `T` will be always DateTime type ? if so, can you please assert for that?

Reply:
can this help better here [https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.as?view=net-6.0#system-runtime-compilerservices-unsafe-as-2(-0](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.as?view=net-6.0#system-runtime-compilerservices-unsafe-as-2(-0)@)?

In general will be good if we can avoid the boxing here.

Reply:
Apologies, this was leftover from an incorrect comparison of DateTime and Bool column. This isn't actually necessary for the DateTime column and I'm removing it.